packages/rescript-relay/src/RescriptRelay.resi

```diff
@@ -635,85 +635,88 @@ module ReadOnlyRecordSourceProxy: {
-    storageKey: Nullable.t<string>,
+    storageKey: nullable<string>,
+    args: nullable<array<normalizationArgument>>,
+    concreteType: nullable<string>,
+    plural: bool,
+    selections: array<JSON.t>,
   }
 
+  @tag("kind")
+  type rec t =
+    | @as("scalar")
+    Scalar({
+        handle: (
+          normalizationScalarField,
+          nullable<'record>,
+          'args,
+          ReadOnlyRecordSourceProxy.t,
+        ) => 'scalarValue,
+      }): t
+    | @as("linked")
+    Linked({
+        handle: (
+          normalizationLinkedField,
+          nullable<'record>,
+          'args,
+          ReadOnlyRecordSourceProxy.t,
+        ) => option<dataId>,
+      }): t
+    | @as("pluralLinked")
+    PluralLinked({
+        handle: (
+          normalizationLinkedField,
+          nullable<'record>,
+          'args,
+          ReadOnlyRecordSourceProxy.t,
+        ) => array<dataId>,
+      }): t
+
   /**Make a `MissingFieldHandler.t` for scalar fields. Give this a handler function that returns `Js.null` (to indicate that data exists but is null), `Js.undefined` (to indicate data is still missing), or a scalar value (to indicate that the value exists even though it's not in the cache, and is the value you send back).*/
   let makeScalarMissingFieldHandler: (
     (
       normalizationScalarField,
-      Nullable.t<'record>,
+      nullable<'record>,
       'args,
       ReadOnlyRecordSourceProxy.t,
     ) => 'scalarValue
   ) => t
 
-  type normalizationLinkedField = {
-    alias: Nullable.t<string>,
-    name: string,
-    storageKey: Nullable.t<string>,
-    args: Nullable.t<array<normalizationArgument>>,
-    concreteType: Nullable.t<string>,
-    plural: bool,
-    selections: array<JSON.t>,
-  }
-
   /**Make a `MissingFieldHandler.t` for linked fields (other objects/records). Give this a handler function that returns `Js.null` (to indicate that the link exists but the linked record is null), `Js.undefined` (to indicate data is still missing), or a `dataId` of the record that is linked at this field.*/
   let makeLinkedMissingFieldHandler: (
     (
       normalizationLinkedField,
-      Nullable.t<RecordProxy.t>,
+      nullable<RecordProxy.t>,
       'args,
       ReadOnlyRecordSourceProxy.t,
-    ) => Nullable.t<dataId>
+    ) => option<dataId>
   ) => t
 
   /**Make a `MissingFieldHandler.t` for lists of linked fields (other objects/records). Give this a handler function that returns `Js.null` (to indicate that the link exists but the linked record is null), `Js.undefined` (to indicate data is still missing), or an array of `Js.Nullable.t<dataId>` where the `dataId`'s are the linked records/objects.*/
   let makePluralLinkedMissingFieldHandler: (
     (
       normalizationLinkedField,
-      Nullable.t<RecordProxy.t>,
+      nullable<RecordProxy.t>,
       'args,
       ReadOnlyRecordSourceProxy.t,
-    ) => Nullable.t<array<Nullable.t<dataId>>>
+    ) => array<dataId>
```

Comment:
Same here, this isn't correct. With this change there's no way to mark things as data missing or data fetched but missing.